compiler/modules/front/stmt.bal

```diff
@@ -62,14 +48,13 @@ type LExprEffect record {|
 
 type LoopContext record {|
     // number of first register created in the loop, including the ones created in the cond
-    int startRegister;
     bir:BasicBlock? onBreak;
     bir:BasicBlock? onContinue;
     LoopContext? enclosing;
-    // will use this with while true to determine whether
-    // following block is reachable
-    Assignment[] onBreakAssignments = [];
-    Assignment[] onContinueAssignments = [];
+    // Will use this to validate assignments in loops and
+    // determine assignments to be passed up
+    BindingChain?[] onBreakBindings = [];
```

Comment:
I find this a bit weird.  Why do we need a BindingChain here?  Is will always be a BindingChain starting with a Binding that represents an assignment, right? It is possible to have a subtype to capture this.